mqtt/mqtt-bridge/src/bridge.rs

```diff
@@ -69,18 +72,24 @@ impl BridgeHandle {
 }
 
 /// Bridge implementation that connects to local broker and remote broker and handles messages flow
-pub struct Bridge<S> {
+pub struct Bridge<S>
+where
+    S: StreamWakeableState + Send + Sync,
+{
     local_pump: Pump<S, LocalUpstreamMqttEventHandler<S>, LocalUpstreamPumpEventHandler>,
     remote_pump: Pump<S, RemoteUpstreamMqttEventHandler<S>, RemoteUpstreamPumpEventHandler>,
 }
 
-impl Bridge<WakingMemoryStore> {
+impl Bridge<RingBuffer> {
     pub fn new_upstream(
         system_address: &str,
         device_id: &str,
         settings: &ConnectionSettings,
     ) -> Result<Self, BridgeError> {
         const BATCH_SIZE: usize = 10;
+        const FILE_NAME: &'static str = "ring_buffer.txt";
+        const FLUSH_OPTIONS: FlushOptions = FlushOptions::AfterXBytes(1024);
+        const MAX_FILE_SIZE: usize = 1024 * 1024;
```

Comment:
All of these default settings here and within the builder should probably go away in favour of some sort of settings or config. Need to look into what to edit to pass along such fields/values.